async-http-client-backend/fs2/src/main/scala/sttp/client/asynchttpclient/fs2/Fs2WebSocketHandler.scala

```diff
@@ -0,0 +1,50 @@
+package sttp.client.asynchttpclient.fs2
+
+import cats.effect._
+import cats.implicits._
+import fs2.concurrent.InspectableQueue
+import sttp.client.asynchttpclient.WebSocketHandler
+import sttp.client.asynchttpclient.internal.{AsyncQueue, NativeWebSocketHandler}
+import sttp.client.impl.cats.CatsMonadAsyncError
+import sttp.client.ws.{WebSocket, WebSocketEvent}
+import sttp.model.ws.WebSocketBufferFull
+
+object Fs2WebSocketHandler {
+  private class Fs2AsyncQueue[F[_], A](queue: InspectableQueue[F, A])(implicit F: Effect[F]) extends AsyncQueue[F, A] {
+    override def clear(): Unit =
+      F.runAsync(queue.getSize.flatMap { size =>
```

Comment:
This approach is *not * generally safe, but given how it is called and executed in this very special setup, it should work as expected. Feel free to challenge though!